### PR TITLE
PROXY-1045 removed auth on hot-reload

### PR DESCRIPTION
**Pull request by bohdan-titan**

### CODE REVIEW

File: src/store/reducers/contracts.js [hunk 1]
```diff
@@ -68,6 +68,17 @@ const reducer = handleActions(
       };
     },
 
+    'purchase-contract-failed': (state, { payload }) => {
+      let array = state.actives;
+      let objIndex = array.findIndex(obj => obj.id == payload.id);
```
Comment (alex-sandrk) [inline src/store/reducers/contracts.js:73]:
```suggestion
      const array = state.actives;
      const objIndex = array.findIndex(obj => obj.id == payload.id);
```

File: public/config/index.js [hunk 2]
```diff
@@ -24,7 +24,9 @@ const chain = {
   buyerWebPort: process.env.BUYER_WEB_DEFAULT_PORT || 8082,
 
   localSellerProxyRouterUrl: `http://localhost:${process.env.SELLER_WEB_DEFAULT_PORT || 8081}`,
-  localBuyerProxyRouterUrl: `http://localhost:${process.env.BUYER_WEB_DEFAULT_PORT || 8082}`
+  localBuyerProxyRouterUrl: `http://localhost:${process.env.BUYER_WEB_DEFAULT_PORT || 8082}`,
+
+  ignoreAuth: process.env.IGNORE_AUTH || false
```
Comment (lsheva) [inline public/config/index.js:29]: bypassAuth sounds better for me

File: src/components/common/Root.js [hunk 3]
```diff
@@ -33,6 +33,14 @@ class Root extends React.Component {
         });
         this.setState({ onboardingComplete });
       })
+      .then(() => {
+        if (this.props.isIgnoreAuth === 'true') {
```
Comment (lsheva) [inline src/components/common/Root.js:37]: isAuthBypassed

**Review by lsheva — APPROVED** [review]
would like to change naming if you dont mind

File: src/components/common/Root.js (hunk 3 above)
Comment (lsheva) [reply src/components/common/Root.js:37]: should be boolean here

File: public/config/index.js (hunk 2 above)
Comment (lsheva) [inline public/config/index.js:29]:
```suggestion
  ignoreAuth: process.env.IGNORE_AUTH === 'true'
```

File: src/components/common/Root.js (hunk 3 above)
Comment (lsheva) [inline src/components/common/Root.js:37]:
```suggestion
        if (this.props.isIgnoreAuth) {
```

**Review by lsheva — CHANGES_REQUESTED** [review]
taking back my approve, see comments

File: src/store/reducers/contracts.js [hunk 4]
```diff
@@ -68,6 +68,17 @@ const reducer = handleActions(
...
+      array[objIndex].inProgress = false;
+      array[objIndex].buyer = '0x0000000000000000000000000000000000000000';
```
Comment (lsheva) [inline src/store/reducers/contracts.js:75]: i would move it to constants, like ZERO_ADDRESS

but maybe there is a better way like `new ContractAddress().toString()`

**Review by lsheva — APPROVED** [review]
excellent job @bohdan-titan

**Pull request merged**